securedrop_client/storage.py

```diff
@@ -559,12 +565,11 @@ def update_replies(
 
             del local_replies_by_uuid[reply.uuid]
             logger.debug("Updated reply {}".format(reply.uuid))
-        elif local_reply and reply.source_uuid in skip_uuids_deleted_conversation:
+        elif reply.source_uuid in skip_uuids_deleted_conversation:
```

Comment:
if there's no local submission, _then_ see if the remote reply belongs to a locally-deleted conversation

Reply:
Okay, yes, your changes make sense (and inline github comments made it much easier to follow, thanks!). Only skip if the "new" reply from the server is one that we deleted locally (if a local reply exists then we obviously didn't delete it locally). Could there be a scenario where `skip_uuids_deleted_conversation` contains a source uuid before all the items in the source conversation have finished being deleted locally? I'm realizing that I need to revisit the code in `_delete_source_collection_from_db` in `storage.py` and do more testing around deletion now that I see some of this logic has changed.